### DISCUSSION TITLE
[Question] Which is the minimum required OpenGL version?

### DISCUSSION THREAD

**Discussion by jrugia** [diasurgical/DevilutionX#4013] [Q&A]
As title says I'd like to know which is the bare minimum OpenGL version required to run this project.

Basically I've acquired an ancient laptop which I'm restoring (Intel Atom N270 / Intel GMA 950 GPU). It supports only up to OpenGL 1.4 though. Thanks for any input :)

**Comment by StephenCWills** [comment]
I haven't been able to find anything in SDL's documentation. You may just have to try it and see. Also, if you find that OpenGL 1.4 doesn't work with SDL2, we also support builds with SDL1.2 so you could try that as well.

**Comment by AJenbo** [comment]
OpenGL isn't required. You can simply disable upscaling in the ini file to use a pure software based rendering similar to the original (this is also what you would get by building for SDL1.2, but disabeling upscaling still leave you all the other benefits of SDL2 and is a lot simpler to do).
For the next release we also support software based upscaling but unless you CPU is pretty powerful you are going to want to disable upscaling if you do not have capable hardware acceleration.

Regarding what is required for hardware accelerated upscaling, SDL2 supports all the way back to OpenGL 1.1, but DirectX 9 and DirectX 11 are also options. There are also backends for OpenGL ES 1/2 as well as hardware specific ones.

**Reply by jrugia** [reply]
Thanks for the detailed response, will give this a shot then :)
